akvo/rsr/static/scripts-src/project-main/project-main-partners.js

```diff
@@ -82,9 +89,42 @@ function renderPartnersTab() {
             }
         },
 
+        partnerName: function(partner){
+            /*
+            Return org.long_name if org.name looks like a truncated version of org.long_name
+
+            When importing organisations from IATI, the name field may have been truncated to 25 or
+            (more seldom) 40 characters.
+            Therefore we check if the name is exactly 25 or 40 chars long AND is identical to the
+            substring of long_name to that length. If this is the case we use the long_name as there
+            is a very high probability that the name field has been truncated.
+
+            NOTE: this is also done in the Organisation.get_name method
```

Comment:
Yup, I was thinking about that too, but it's yet another addition to an already overloaded "quick fix".